scheduling/tasks.go

```diff
@@ -53,11 +55,16 @@ type TaskMetadata struct {
 	DateCreated time.Time  `json:"dateCreated"`
 }
 
-func (t *TaskMetadata) Marshal() ([]byte, error) {
-	return json.Marshal(t)
+func (t *TaskMetadata) Marshal() ([]byte, fferr.GRPCError) {
```

Comment:
let's use the same receiver name for all these.  goland gives me warnings on these. Does vscode not have a go extension that does static checks/lints on go?

Reply:
look at the reciever names, sometimes you use t, sometimes you use tm